src/katsdpmodels/band_mask.py

```diff
@@ -0,0 +1,102 @@
+################################################################################
+# Copyright (c) 2020, National Research Foundation (SARAO)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Masks for roll-off at the edges of a band."""
+
+from typing import Type, TypeVar, ClassVar, Any
+from typing_extensions import Literal
+
+import numpy as np
+import astropy.units as u
+import astropy.table
+import h5py
+
+from . import models
+
+_B = TypeVar('_B', bound='BandMaskRanges')
+
+
+class Band:
+    """Defines a band for use with :class:`BandMask`."""
+
+    def __init__(self, bandwidth: u.Quantity, centre_frequency: u.Quantity):
+        self.bandwidth = bandwidth.to(u.Hz)
+        self.centre_frequency = centre_frequency.to(u.Hz)
+
+    @property
+    def min_frequency(self) -> u.Quantity:
+        return self.centre_frequency - self.bandwidth / 2
+
+    @property
+    def max_frequency(self) -> u.Quantity:
+        return self.centre_frequency + self.bandwidth / 2
+
+
+class BandMask(models.SimpleHDF5Model):
+    model_type: ClassVar[Literal['band_mask']] = 'band_mask'
+
+    def is_masked(self, band: Band, frequency: u.Quantity) -> Any:
+        raise NotImplementedError()      # pragma: nocover
+
+    @classmethod
+    def from_hdf5(cls, hdf5: h5py.File) -> 'BandMask':
+        model_format = models.get_hdf5_attr(hdf5.attrs, 'model_format', str) or ''
+        if model_format == 'ranges':
```

Comment:
I have been wondering about the model_format and how it connects to the model class - I would have expected more connective syntax between BandMask and 'ranges'. Not that we have to match the model_format and class names, but the question is why not ?

Reply:
BandMask is the base class corresponding to model_type=='band_mask'. BandMaskRanges is the class that corresponds to model_type=='band_mask' && model_format=='ranges'. Does that make more sense?

Reply:
Doh - brain skipped over the model_type definition. Type + Format makes sense...